go oops/oops: Truncate logs that have more than 500 lines

CarlaBerteil opened the pull request:
Some logs are really long wich makes it really difficult to read and diagnose
issues.
This change created an API that will print the number of frames requested. 

I tested those chages locally, with a limit of 1  frame: 
![truncate_logs](https://user-images.githubusercontent.com/50117085/67886317-81594e80-fb06-11e9-823a-f5d2d6763bbc.gif)

anubhav94 commented on oops/oops.go:
```diff
@@ -180,6 +182,11 @@ func (e *oopsError) writeStackTrace(w io.Writer) {
 			}
 			fmt.Fprintf(w, "\t%s:%d\n", frame.File, frame.Line)
+			lineCount++
+            if lineCount > maxStackLength {
```
the indentation here is weird, can we fix that?

CarlaBerteil replied:
done

anubhav94 reviewed (approved):
LGTM - wondering if we should do this during stack generation or have the full stack and truncate before outputting. let's see what @stephen and @jellevandenhooff think!

stephen reviewed (changes requested):
would prefer we figure out how to we get emitters to do this instead of changing the core library.

alternatively, consider printing only the top stack frame instead of all frames?

CarlaBerteil committed:
go oops/error: Add a method that only prints a number of frames

We want to be able to print only the top frames so we don't crowd the logs.
This commit refactors `writeStacktrace` into a separate function so we can call
it for the new `ErrorFrame` function, that will print the top n frames of the
stack.

anubhav94 commented on oops/oops.go:
```diff
@@ -38,6 +38,17 @@ func (e *oopsError) Error() string {
+func (e *oopsError) ErrorFrame(n int) string {
+	var buffer bytes.Buffer
+	fmt.Fprintf(&buffer, "%s\n\n", e..Error())
+	framesLenght := len(Frames(e))
```
typo: /s/framesLenght/framesLength/g

anubhav94 commented on oops/oops.go:
```diff
@@ -171,16 +182,19 @@ func (e *oopsError) writeStackTrace(w io.Writer) {
-				fmt.Fprintf(w, "%s\n", frame.Function)
-			}
-			fmt.Fprintf(w, "\t%s:%d\n", frame.File, frame.Line)
+func writeFrameTrace(w io.Writer, frames []Frame) {
```
nit: it would be good to mention that we are refactoring some part of `writeStackTrace` into a separate function so we can call it from a new method to print the topN frames of the stack etc

CarlaBerteil replied:
Done

CarlaBerteil commented:
@stephen  I made some changes to this. Could you please review and let me know if that is better?

ianvernon commented:
@CarlaBerteil any particular reason why you closed this?
